### PR TITLE
build: migrate discord4j to JDA

### PR DESCRIPTION
**Pull request by Tran-Antoine**

### CODE REVIEW

**Comment by Tran-Antoine** [comment]
All the dependencies have been migrated, however, the bot hasn't been entirely tested yet, so there might still be some bugs (which is the reason why this is a draft)

**Comment by Tran-Antoine** [comment]
Side not: it seems like I have pushed the token already... *sighs*
The token is safely put in an environment variable on Heroku, but for the sake of local testing, I always run the code with the token copy-pasted and then forget to remove it before pushing. Token refreshed now so that one isn't valid anymore (obviously)

**Comment by Tran-Antoine** [comment]
Ok I think I've tested the bot enough, all discord-related behaviors have been tested and seem to be working

File: discord-impl/src/main/java/net/starype/quiz/discordimpl/core/BotMain.java [hunk 1]
```diff
@@ -1,39 +1,26 @@
...
-        dispatcher
-                .on(MessageCreateEvent.class)
-                .filter(MessageInputListener.createFilter())
-                .retry()
-                .subscribe(new MessageInputListener(new LobbyList(reactionListener), new GameList()));
-        client.onDisconnect().block();
+        JDABuilder builder = JDABuilder.createDefault(System.getenv("BOT_TOKEN"));
+        JDA jda = builder.disableCache(CacheFlag.MEMBER_OVERRIDES, CacheFlag.VOICE_STATE)
+                .setBulkDeleteSplittingEnabled(false)
+                .setActivity(Activity.playing("supervising the quiz world championship"))
```
Comment (mredstone6) [inline discord-impl/src/main/java/net/starype/quiz/discordimpl/core/BotMain.java:19]: This name is illegal

**Review by mredstone6 — APPROVED** [review]

File: discord-impl/src/main/java/net/starype/quiz/discordimpl/core/BotMain.java (hunk 1 above)
Comment (mredstone6) [reply discord-impl/src/main/java/net/starype/quiz/discordimpl/core/BotMain.java:19]: "supervising the quiz world championship"

Comment (Tran-Antoine) [reply discord-impl/src/main/java/net/starype/quiz/discordimpl/core/BotMain.java:19]: So is your nickname

**Review by mredstone6 — CHANGES_REQUESTED** [review]
Have you heard about the discord API for java named Javacord ?

File: discord-impl/src/main/java/net/starype/quiz/discordimpl/core/BotMain.java (hunk 1 above)
Comment (mredstone6) [reply discord-impl/src/main/java/net/starype/quiz/discordimpl/core/BotMain.java:19]: This PR could have been merged before 5 month

**Review by mredstone6 — APPROVED** [review]

**Pull request merged**